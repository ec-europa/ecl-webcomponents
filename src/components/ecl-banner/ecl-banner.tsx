import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-banner',
  styleUrls: {
    ec: './build/styles/ecl-banner-ec.css',
    eu: './build/styles/ecl-banner-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclBanner {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() external: boolean = false;
  @Prop() variant: string = 'display';
  @Prop() fullWidth: boolean = false;
  @Prop() bannerTitle: string;
  @Prop() image: string;
  @Prop() ctaLabel: string;
  @Prop() ctaLink: string;
  @Prop() centered: boolean = true;
  @Prop() credit: string;
  @Prop() size: string = 'm';
  @Prop() ctaClicked: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-banner`,
      `ecl-banner--${this.variant}`,
      `ecl-banner--${this.size}`,
      this.styleClass
    ];

    if (this.centered) {
      styleClasses.push('ecl-banner--centered');
    }
    if (this.fullWidth) {
      styleClasses.push('ecl-banner--full-width');
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    this.el.querySelector('.ecl-banner__cta').addEventListener('click', () => {
      this.ctaClicked = true;
    });
  }

  render() {
    const imageBanners = [
      'text-box',
      'image-overlay',
      'text-highlight',
    ];
    return (
      <section class={this.getClass()}>
      { imageBanners.includes(this.variant) && this.image ?
        <div
          class="ecl-banner__image"
          style={{ backgroundImage: "url(" + this.image + ")"}}
        >
        </div>
        : '' }
      { this.credit && this.variant != 'plain-background' ?
        <div class="ecl-banner__credit">{this.credit}</div> : ''
      }
        <div class="ecl-container">
          <div class="ecl-banner__container">
            <div class="ecl-banner__content">
            { this.bannerTitle ?
              <div class="ecl-banner__title">
                <span class="ecl-banner__title-text">{this.bannerTitle}</span>
              </div> : ''
            }
              <p class="ecl-banner__description">
                <span class="ecl-banner__description-text">
                  <slot></slot>
                </span>
              </p>
            { this.ctaLabel && this.ctaLink ?
              <div class="ecl-banner__cta">
                <ecl-link
                  variant="cta"
                  path={this.ctaLink}
                  style-class={`ecl-banner__link-cta sc-ecl-banner-${this.theme}`}
                >
                  {this.ctaLabel}
                  <ecl-icon
                    icon="corner-arrow"
                    slot="icon-after"
                    transform="rotate-90"
                  ></ecl-icon>
                </ecl-link>
              </div>
            : '' }
            </div>
          </div>
        </div>
      </section>
    );
  }
}
