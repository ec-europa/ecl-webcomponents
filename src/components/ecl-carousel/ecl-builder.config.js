const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-carousel/carousel.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-carousel-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-carousel-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-carousel-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-carousel-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-carousel-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
