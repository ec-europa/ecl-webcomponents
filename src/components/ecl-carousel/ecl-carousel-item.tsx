import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-carousel-item',
  shadow: false,
})
export class EclCarouselItem {
  @Prop() styleClass: string = '';
  @Prop() bannerTitle: string = '';
  @Prop() bannerVariant: string = '';
  @Prop() image: string = '';
  @Prop() theme: string = 'ec';
  @Prop() ctaLink: string;
  @Prop() ctaLabel: string;
  @Prop() size: string = 'm';
  @Prop() ariaLabel: string;
  @Prop() centered: boolean;

  getClass(): string {
    return [
      `ecl-carousel__item`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <div
        class="ecl-carousel__slide"
        role="group"
        aria-label={this.ariaLabel}
      >
        <ecl-banner
          banner-title={this.bannerTitle}
          image={this.image}
          variant={this.bannerVariant}
          theme={this.theme}
          cta-link={this.ctaLink}
          cta-label={this.ctaLabel}
          centered={this.centered}
          size={this.size}
        >
          <slot></slot>
        </ecl-banner>
      </div>
    );
  }
}
  