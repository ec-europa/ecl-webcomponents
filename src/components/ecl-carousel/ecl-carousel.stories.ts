const getArgs = () => {
  return {
    size: 'm',
  };
};

const getArgTypes = () => {
  return {
    size: {
      type: { name: 'select'},
      options: [
        's', 'm', 'l',
      ],
      description: 'Size of the carousel (s, m, l)',
    },
  };
};

export default {
  title: 'Components/carousel',
};

const Template = (args) => 
  `<ecl-carousel
    carousel-id="ecl-carousel-demo"
    ecl-script
    slides-number=3
  >
    <ecl-carousel-item
      banner-variant="text-box"
      theme="${args.theme}"
      cta-link="/example"
      cta-label="Subscribe"
      centered=false
      size="${args.size}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
      banner-title="Duis vitae pulvinar turpis"
    >
      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
    </ecl-carousel-item>
    <ecl-carousel-item
      banner-variant="text-highlight"
      theme="${args.theme}"
      cta-link="/example"
      cta-label="Subscribe"
      centered=false
      size="${args.size}"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
      banner-title="Duis vitae pulvinar turpis"
    >
      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
    </ecl-carousel-item>
    <ecl-carousel-item
      banner-variant="image-overlay"
      theme="${args.theme}"
      cta-link="/example"
      size="${args.size}"
      cta-label="Subscribe"
      image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
      banner-title="Duis vitae pulvinar turpis"
    >
      Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
    </ecl-carousel-item>
  </ecl-carousel>`;

export const Carousel = Template.bind({});
Carousel.storyName = 'default';
Carousel.args = getArgs();
Carousel.argTypes = getArgTypes();
