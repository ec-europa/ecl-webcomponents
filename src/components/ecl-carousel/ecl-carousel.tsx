import { Component, h, Prop, Element, getAssetPath } from '@stencil/core';
declare const ECL: any;

@Component({
  tag: 'ecl-carousel',
  styleUrls: {
    ec: './build/styles/ecl-carousel-ec.css',
    eu: './build/styles/ecl-carousel-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclCarousel {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() eclScript: boolean = false;
  @Prop() carouselId: string;
  @Prop() slidesNumber: number;
  @Prop() counterLabel: string = 'of';
  @Prop() srNavigation: string = 'Go to slide %d';
  @Prop() srPrevious: string = 'Prev slides';
  @Prop() srNext: string = 'Next slides';
  @Prop() srPause: string = 'Pause carousel';
  @Prop() srPlay: string = 'Play carousel';

  getClass(): string {
    return [
      `ecl-carousel`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-carousel-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const carousel = new ECL.Carousel(this.el.firstElementChild);
        carousel.init();
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <div
        class={this.getClass()}
        data-ecl-auto-init="Carousel"
      >
        <div class="ecl-carousel__container">
          <div
            class="ecl-carousel__slides"
            id={this.carouselId}
          >
            <slot></slot>
          </div>
          <button 
            class="ecl-carousel__prev"
          >
            <ecl-icon
              icon="corner-arrow"
              transform="rotate-270"
              style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec"
            ></ecl-icon>
             <span class="ecl-u-sr-only">{this.srPrevious}</span>
          </button>
          <button class="ecl-carousel__next">
            <ecl-icon
              icon="corner-arrow"
              transform="rotate-90"
              style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec"
            ></ecl-icon>
            <span class="ecl-u-sr-only">{this.srNext}</span>
          </button>
        </div>
        <div class="ecl-carousel__controls">
          <div class="ecl-container">
            <div class="ecl-carousel__autoplay">
              <button class="ecl-carousel__play">
                <ecl-icon
                  icon="play"
                  size="l"
                  color="inverted"
                  style-class={`ecl-carousel__icon-default sc-ecl-carousel-${this.theme}`}
                ></ecl-icon>
                <ecl-icon
                  icon="play-filled"
                  size="l"
                  color="inverted"
                  style-class={`ecl-carousel__icon-active sc-ecl-carousel-${this.theme}`}
                ></ecl-icon>
                <span class="ecl-u-sr-only">{this.srPlay}</span>
              </button>
              <button class="ecl-carousel__pause">
                <ecl-icon
                  icon="pause"
                  size="l"
                  style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec"
                ></ecl-icon>
                <ecl-icon
                  icon="pause-filled"
                  size="l"
                  color="inverted"
                  style-class={`ecl-carousel__icon-active sc-ecl-carousel-${this.theme}`}
                ></ecl-icon>
                <span class="ecl-u-sr-only">{this.srPause}</span>
              </button>
            </div>
            <div class="ecl-carousel__navigation">
            { [...Array(this.slidesNumber)].map((i) =>
              <button class="ecl-carousel__navigation-item">
                <span class="ecl-u-sr-only">{this.srNavigation.replace('%d', i)}</span>
              </button>
            ) } 
            </div>
            <div class="ecl-carousel__pagination">
              <span class="ecl-carousel__current"></span> {this.counterLabel} <span class="ecl-carousel__max">{this.slidesNumber}</span>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
