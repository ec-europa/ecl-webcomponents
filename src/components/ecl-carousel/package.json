{
  "name": "@ecl/web-components-carousel",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.3.0",
  "description": "ECL Carousel",
  "main": "ecl-carousel.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env NODE_ENV=production ecl-builder styles",
    "build:scripts": "cross-env ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.1",
    "@ecl/resources-ec-icons": "3.7.1",
    "@ecl/resources-eu-icons": "3.7.1",
    "@ecl/vanilla-component-carousel": "3.7.1",
    "@ecl/vanilla-layout-grid": "3.7.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
