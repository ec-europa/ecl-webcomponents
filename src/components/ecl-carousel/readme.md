# ecl-carousel



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type      | Default     |
| --------------- | ---------------- | ----------- | --------- | ----------- |
| `ariaLabel`     | `aria-label`     |             | `string`  | `undefined` |
| `bannerTitle`   | `banner-title`   |             | `string`  | `''`        |
| `bannerVariant` | `banner-variant` |             | `string`  | `''`        |
| `centered`      | `centered`       |             | `boolean` | `undefined` |
| `ctaLabel`      | `cta-label`      |             | `string`  | `undefined` |
| `ctaLink`       | `cta-link`       |             | `string`  | `undefined` |
| `image`         | `image`          |             | `string`  | `''`        |
| `size`          | `size`           |             | `string`  | `'m'`       |
| `styleClass`    | `style-class`    |             | `string`  | `''`        |
| `theme`         | `theme`          |             | `string`  | `'ec'`      |


## Dependencies

### Depends on

- [ecl-banner](../ecl-banner)

### Graph
```mermaid
graph TD;
  ecl-carousel-item --> ecl-banner
  ecl-banner --> ecl-picture
  ecl-banner --> ecl-link
  ecl-banner --> ecl-icon
  ecl-link --> ecl-icon
  ecl-link --> ecl-indicator
  style ecl-carousel-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
