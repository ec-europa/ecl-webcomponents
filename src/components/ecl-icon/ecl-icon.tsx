import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}