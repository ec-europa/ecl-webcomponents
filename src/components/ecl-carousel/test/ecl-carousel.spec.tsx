import { newSpecPage } from '@stencil/core/testing';
import { EclCarousel } from '../ecl-carousel';
import { EclBanner } from '../../ecl-banner/ecl-banner';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-carousel', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclCarousel, EclBanner, EclIcon],
      html: `<ecl-carousel
              carousel-id="ecl-carousel-demo"
              ecl-script
              slides-number="3"
            >
              <ecl-carousel-item
                banner-variant="text-box"
                theme="ec"
                cta-link="/example.html"
                cta-label="Subscribe"
                centered=false
                size="l"
                image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg"
                banner-title="Duis vitae pulvinar turpis"
              >
                Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
              </ecl-carousel-item>
              <ecl-carousel-item
                banner-variant="text-highlight"
                theme="ec"
                size="l"
                cta-link="/example.html"
                cta-label="Subscribe"
                centered=false
                image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg"
                banner-title="Duis vitae pulvinar turpis"
              >
                Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
              </ecl-carousel-item>
              <ecl-carousel-item
                banner-variant="image-overlay"
                theme="ec"
                size="l"
                cta-link="/example.html"
                cta-label="Subscribe"
                image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg"
                banner-title="Duis vitae pulvinar turpis"
              >
                Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
              </ecl-carousel-item>
            </ecl-carousel>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
