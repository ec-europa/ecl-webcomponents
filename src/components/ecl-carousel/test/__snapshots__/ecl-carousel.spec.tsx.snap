// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-carousel renders 1`] = `
<ecl-carousel carousel-id="ecl-carousel-demo" ecl-script="" slides-number="3">
  <!---->
  <div class="ecl-carousel" data-ecl-carosuel>
    <div class="ecl-carousel__container">
      <div class="ecl-carousel__slides" id="ecl-carousel-demo">
        <ecl-carousel-item banner-title="Duis vitae pulvinar turpis" banner-variant="text-box" centered="false" cta-label="Subscribe" cta-link="/example" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" size="l" theme="ec">
          Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
        </ecl-carousel-item>
        <ecl-carousel-item banner-title="Duis vitae pulvinar turpis" banner-variant="text-highlight" centered="false" cta-label="Subscribe" cta-link="/example" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" size="l" theme="ec">
          Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
        </ecl-carousel-item>
        <ecl-carousel-item banner-title="Duis vitae pulvinar turpis" banner-variant="image-overlay" cta-label="Subscribe" cta-link="/example" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" size="l" theme="ec">
          Integer quis lorem tellus. Nullam sollicitudin suscipit diam, ac blandit ipsum tempor consectetur
        </ecl-carousel-item>
      </div>
      <button class="ecl-carousel__prev">
        <ecl-icon style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec">
          <svg class="ecl-carousel__icon-default ecl-icon ecl-icon--inverted ecl-icon--rotate-270 ecl-icon--xs sc-ecl-carousel-ec">
            <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
          </svg>
        </ecl-icon>
        <span class="ecl-u-sr-only">
          Prev slides
        </span>
      </button>
      <button class="ecl-carousel__next">
        <ecl-icon style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec">
          <svg class="ecl-carousel__icon-default ecl-icon ecl-icon--inverted ecl-icon--rotate-90 ecl-icon--xs sc-ecl-carousel-ec">
            <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
          </svg>
        </ecl-icon>
        <span class="ecl-u-sr-only">
          Next slides
        </span>
      </button>
    </div>
    <div class="ecl-carousel__controls">
      <div class="ecl-container">
        <div class="ecl-carousel__autoplay">
          <button class="ecl-carousel__play">
            <ecl-icon style-class="ecl-carousel__icon-default sc-ecl-carousel-ec">
              <svg class="ecl-carousel__icon-default ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-carousel-ec">
                <use xlink:href="/build/images/ec/icons.svg#play"></use>
              </svg>
            </ecl-icon>
            <ecl-icon style-class="ecl-carousel__icon-active sc-ecl-carousel-ec">
              <svg class="ecl-carousel__icon-active ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-carousel-ec">
                <use xlink:href="/build/images/ec/icons.svg#play-filled"></use>
              </svg>
            </ecl-icon>
            <span class="ecl-u-sr-only">
              Play carousel
            </span>
          </button>
          <button class="ecl-carousel__pause">
            <ecl-icon style-class="ecl-icon--inverted ecl-carousel__icon-default sc-ecl-carousel-ec">
              <svg class="ecl-carousel__icon-default ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-carousel-ec">
                <use xlink:href="/build/images/ec/icons.svg#pause"></use>
              </svg>
            </ecl-icon>
            <ecl-icon style-class="ecl-carousel__icon-active sc-ecl-carousel-ec">
              <svg class="ecl-carousel__icon-active ecl-icon ecl-icon--inverted ecl-icon--l sc-ecl-carousel-ec">
                <use xlink:href="/build/images/ec/icons.svg#pause-filled"></use>
              </svg>
            </ecl-icon>
            <span class="ecl-u-sr-only">
              Pause carousel
            </span>
          </button>
        </div>
        <div class="ecl-carousel__navigation">
          <button class="ecl-carousel__navigation-item">
            <span class="ecl-u-sr-only">
              Go to slide undefined
            </span>
          </button>
          <button class="ecl-carousel__navigation-item">
            <span class="ecl-u-sr-only">
              Go to slide undefined
            </span>
          </button>
          <button class="ecl-carousel__navigation-item">
            <span class="ecl-u-sr-only">
              Go to slide undefined
            </span>
          </button>
        </div>
        <div class="ecl-carousel__pagination">
          <span class="ecl-carousel__current"></span>
          of
          <span class="ecl-carousel__max">
            3
          </span>
        </div>
      </div>
    </div>
  </div>
</ecl-carousel>
`;
